File: 프로그래머스/0/120835. 진료 순서 정하기/진료 순서 정하기.java
import java.util.*;
class Solution {
    public int[] solution(int[] emergency) {
        int[] answer = new int[emergency.length];
        
        int[] rank = emergency.clone();
        
        Arrays.sort(rank);
        
        for(int i = 0; i < emergency.length; i++){
            for(int j = 0; j < rank.length; j++){
                if(emergency[i] == rank[j]) answer[i] = rank.length - j;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120835. 진료 순서 정하기/진료 순서 정하기.py
def solution(emergency):
    sort = sorted(emergency, reverse=True)                      # 내림차순 리스트 생성
    rank = {key : value + 1 for value, key in enumerate(sort)}  # 딕셔너리 생성
    answer = [rank[x] for x in emergency]
    
    return answer